_posts/2017-12-19-2017-year-in-review.md

```diff
@@ -0,0 +1,144 @@
+
+![dA Platform 2 architecture overview](https://data-artisans.com/wp-content/uploads/2017/09/application_manager_opt4@3x.png)
+
+We aim for dA Platform 2 to be generally available in the first quarter of 2018, providing a commercialy backed platform to the Flink community from the team who helped create the open source project.
+
+### Flink Ecosystem Growth
+
+Flink was added to a selection of distributions and integrations during 2017, making it easier for a wider user base to get started with Flink:
+
+-   [Official Docker image](https://hub.docker.com/r/_/flink/)
+-   [Official DC/OS and Mesos support](https://ci.apache.org/projects/flink/flink-docs-release-1.3/setup/mesos.html)
+-   [A Flink connector](https://data-artisans.com/blog/dellemc-launches-open-source-pravega-complete-apache-flink-connector) for [Pravega](http://pravega.io), Dell/EMC's streaming storage system.
+-   Uber announced AthenaX, a streaming SQL platform [powered by Apache Flink](https://data-artisans.com/blog/uber-introduces-open-source-athenax-streaming-sql-platform-apache-flink).
+
+### Feature Timeline in 2017
+
+Just in time for the end of the year, [our 1.4 release](read the full release announcement](<http://flink.apache.org/news/2017/12/12/release-1.4.0.html>) landed in mid-December culminating 5 months of work and the resolution of more than 900 issues. This is the fifth major release in the 1.x.y series.
+
+Here's a selection of major features added to Flink over the course of 2017:
+
+![Flink Release Timeline 2017]({{ site.baseurl }}/img/blog/flink-releases-2017.png)
+
+<!-- TODO: Is the number below correct? It doesn't seem that high -->
+
+If you take a look at [the resolved issues for 2017 on Jira](https://issues.apache.org/jira/browse/FLINK-4861?jql=project%20%3D%20FLINK%20AND%20issuetype%20%3D%20%22New%20Feature%22%20AND%20status%20in%20(Resolved%2C%20Closed)%20AND%20resolved%20%3E%3D%202017-01-01%20AND%20resolved%20%3C%3D%202017-12-31%20ORDER%20BY%20resolved%20ASC) you can see that the community resolved over 100 issues and feature additions.
+
+In terms of roadmap commitments from 2016, there is mixed news, with some items a part of current releases, others scheduled for upcoming releases and some that remain under discussion.
+
+## Looking ahead to 2018
+
+A good source of information about the Flink community's roadmap is the list of [Flink Improvement Proposals (FLIPs)](https://cwiki.apache.org/confluence/display/FLINK/Flink+Improvement+Proposals) in the project wiki. Below, we'll highlight a selection of FLIPs that have been accepted by the community as well as some that are still under discussion.
+
+**We should note that work is already underway on a number of these features, and some will even be included in Flink 1.2 at the beginning of 2017.**
```

Comment:
@wints I made up this list from the FLIPS, would be good to get confirmation and then rewrite closing paragraph appropriately.

Reply:
@fhueske would be a good resource on which FLIPs to discuss. Should definitely include FLIP-6, which has been in the works for quite awhile and is hopefully going to make it into Flink 1.5 early next year :)

Reply:
"...some will even be included in Flink 1.5 at the beginning of 2017" (current version says 1.2)

Reply:
Let's add:

**Fine-grained recovery from task failures**, as described in [FLIP-1](https://cwiki.apache.org/confluence/display/FLINK/FLIP-1+%3A+Fine+Grained+Recovery+from+Task+Failures) to improve recovery efficiency and only re-execute failed tasks, reducing the amount of state that needs to be transferred on recovery.